src/docs/release/breaking-changes/use-maxLengthEnforcement-instead-of-maxLengthEnforced.md

```diff
@@ -0,0 +1,140 @@
+---
+title: Use maxLengthEnforcement instead of maxLengthEnforced
+description: Introducing the MaxLengthEnforcement enum.
+---
+
+## Summary
+
+_If you don't use the `maxLength` or `maxLengthEnforced` properties in a
+`TextField`, `TextFormField`, or `CupertinoTextField`,
+or you don't care about the truncation behavior of an
+editable text field, you can stop reading._
+
+To control the behavior of `maxLength`
+in the `LengthLimitingTextInputFormatter`,
+use `maxLengthEnforcement` instead of `maxLengthEnforced`, which is deprecated.
+
+## Context
+
+Before [PR 63754][]: Fix TextField crashed with composing and maxLength set,
+the `LengthLimitingTextInputFormatter` always truncated the value in the text
+field, which resulted in engine crashes when the composing range sometimes
+overflowed. With this change, an editable `TextEditingValue` is not truncated
+if only `maxLength` is set, but _is_ truncated after editing is complete.
+
+With [PR 68086][]: Introduce `MaxLengthEnforcement`,
+the truncation behavior is now controlled by passing
+`maxLengthLimitingEnforcement` to the `LengthLimitingTextInputFormatter`,
+which is then exposed to the `TextField`, `TextFormField`,
+and `CupertinoTextField` widgets.
+The `maxLengthEnforced` property is now deprecated.
+(For more information, see the docs for [`maxLength`][] and
+[`MaxLengthEnforcement`][].)
+
+The default max length truncation behavior, `MaxLengthEnforcement.enforced`,
+always truncates.
+
+## Description of change
+
+* Added a `MaxLengthLimitingEnforcement` enum, which controls all max length
+  limiting behavior.
+* The `maxLengthEnforced` property in editable text field classes is deprecated.
+* A named optional parameter, `maxLengthLimitingEnforcement` is added to
+  the `TextField`, `TextFormField`, `CupertinoTextField`,
+  and `LengthLimitingTextInputFormatter` classes, which defaults to
+  `MaxLengthLimitingEnforcement.enforced`.
+
+## Migration guide
+
+The following tips describe how to emit a warning, but not truncate,
+on entering text, and how to truncate as text is entered.
+
+### Not enforces limitation
+
+To print a max length warning in `TextField`, but _not_ truncate when the limit
+is exceed, use `MaxLengthEnforcement.none` instead of
+`maxLengthEnforced: false`.
+
+Code before migration:
+
+<!-- skip -->
+```dart
+TextField(
+  maxLength: 6,
+  maxLengthEnforced: false,
+)
+```
+
+Code after migration:
+
+<!-- skip -->
+```dart
+TextField(
+  maxLength: 6,
+  maxLengthEnforcement: MaxLengthEnforcement.none,
+)
+```
+
+For `CupertinoTextField`, just don't set the `maxLength` value.
```

Comment:
Maybe explain that CupertinoTextField doesn't show error messages?

"For `CupertinoTextField`, which isn't able to show error messages, just don't set the `maxLength` value."